send_later_popover: Redesign popover and add accessibility improvements.

sayamsamal opened the pull request:
As part of the popover menu redesign, this PR redesigns the send later popover using the new "popover-menu" tippy theme and improves accessibility by using appropriate ARIA attributes.

Fixes #25117 (part of #28699.)

[CZO Discussion](https://chat.zulip.org/#narrow/stream/101-design/topic/UI.20redesign.3A.20send.20menu/near/1725684)

> [!NOTE]  
> [This CZO discussion](https://chat.zulip.org/#narrow/stream/101-design/topic/UI.20redesign.3A.20send.20menu/near/1839282) tracks the discussion on the repeated use scheduling and draft related icons in this popover. 

<!-- Describe your pull request here.-->

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

| Desc | Before | After |
|--------|--------|--------|
| Light Mode | <img width="374" alt="send_options_light_before" src="https://github.com/zulip/zulip/assets/82862779/bc51f310-8a29-4a9b-85bc-a2bcb59cf71a"> | <img width="343" alt="Screenshot 2024-07-11 at 1 25 52 AM" src="https://github.com/zulip/zulip/assets/82862779/3d24c973-99dc-4c05-8322-4df8a4bf54da"> |
| Dark Mode | <img width="374" alt="send_options_dark_before" src="https://github.com/zulip/zulip/assets/82862779/08b6789a-85f7-43fe-ba5f-ef5b38df623b"> | <img width="343" alt="Screenshot 2024-07-11 at 1 25 36 AM" src="https://github.com/zulip/zulip/assets/82862779/73e3d96c-6a65-42ac-adc1-0a1094b6ec02"> |

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

sayamsamal commented:
@karlstolley could you please review this one?

timabbott commented:
Looking at the screenshots, this looks considerably nicer, except that the duplicate icons don't look great.

Is it easy to just have the icons be absent for v1? We should ask Vlad for recommendations in #design, but it'd not be a regression to merge without waiting for that. (But definitely not worth doing extra work to make the framework support icon-less entries if that's any effort)

timabbott commented on web/src/drafts.ts:
```diff
@@ -237,6 +237,22 @@ export function update_compose_draft_count(): void {
+    } else {
+        $count_ele.hide();
+    }
+}
```
How does this differ from the previous function? It seems kinda a duplicate... is the issue just that different element names? If so, we should just make `$count_ele` a parameter, maybe calling it `$count_element`.

sayamsamal replied:
The issue was that one unread counter (the one one above the compose box send button), displays the number between two brackets `()` while the other unread counter (the one in the popover menu) is similar to the left sidebar one.

sayamsamal replied:
I matched the structure of the both the count elements, and just removed the brackets `()` from the count element in the popover drafts count.

```
<span class="compose-drafts-count-container"><span class="unread_count compose-drafts-count"></span></span>
```

sayamsamal replied:
Here, the only purpose of `.compose-drafts-count-container` container is to match the structure expected by the `drafts.update_compose_draft_count()` method.

timabbott replied:
Sounds good!

timabbott commented:
I believe this one is still waiting on the icon updates to be pushed.

sayamsamal commented:
## Updates

- Added the new icons for the drafts and schedule related menu options.
- Addressed the duplication of code in the logic to update the drafts counts.

sayamsamal commented:
## Updated Screenshots
<img width="343" alt="Screenshot 2024-07-11 at 1 25 52 AM" src="https://github.com/zulip/zulip/assets/82862779/3d24c973-99dc-4c05-8322-4df8a4bf54da">
<img width="343" alt="Screenshot 2024-07-11 at 1 25 36 AM" src="https://github.com/zulip/zulip/assets/82862779/73e3d96c-6a65-42ac-adc1-0a1094b6ec02">

timabbott commented on web/templates/popovers/send_later_popover.hbs:
```diff
@@ -1,62 +1,68 @@
+                        </span>
+                        <span class="enter_sends_minor enter_sends_choice_text">
+                            Press {{popover_hotkey_hints "Enter"}} to add a new line
+                        </span>
```
These strings seem to have lost their translation tags.

sayamsamal replied:
Ah, my bad. Re-added them like such:
```handlebars
{{#tr}}
    Press <z-shortcut></z-shortcut> to send
    {{#*inline "z-shortcut"}}{{popover_hotkey_hints "Enter"}}{{/inline}}
{{/tr}}
```

timabbott commented on web/styles/compose.css:
```diff
@@ -1001,41 +1001,80 @@ textarea.new_message_textarea {
+            background: var(--color-background-active-popover-menu);
+        }
 
         & input[type="radio"] {
```
This selector should probably just be replaced with `.enter-sends-choice-radio` or something? Generally these element-based selectors are a bad pattern.

sayamsamal replied:
Makes sense! Replaced `input[type="radio"]` selector with `.enter_sends_choice_radio`.

sayamsamal commented:
@timabbott I have addressed the feedback, ready for another review. :)

timabbott commented:
Merged, thanks @sayamsamal!
